apps/console/src/features/roles/components/edit-role/edit-role-permission-updated.tsx

```diff
@@ -0,0 +1,459 @@
+                    message: t("console:manage.features.roles.notifications.fetchAPIResource.error.message")
+                });
+            }
+        }
+        setSelectedAPIResources(apiResources);        
+    };
+
+    const getExistingAPIResources = (): void => {
+        const currentPermissions: SelectedPermissionsInterface[] = [];
+        const uniqueAPIResourceIds: string[] = [];
+
+        roleObject?.permissions?.forEach((permission: RolePermissionInterface | string) => {
+            // Check if permission is of type RolePermissionInterface
+            if (typeof permission !== "string") {
+                const apiResourceId: string = permission["$ref"].split("/")[7];
+
+                // Populate the selected permissions list.
+                if (!currentPermissions.find((selectedPermission: SelectedPermissionsInterface) =>
+                    selectedPermission.apiResourceId === apiResourceId)) {
+                    // Found a new API resource id                                        
+                    uniqueAPIResourceIds.push(apiResourceId);
+                    
+                    // Add it to the selected API resources list.
+                    currentPermissions.push({
+                        apiResourceId: apiResourceId,
+                        scopes: [ {
+                            displayName: permission.display,
+                            id: permission.value,
+                            name: permission.value
+                        } ]
+                    });
+                } else {
+                    // Found a matching API resource id
+                    const index: number = currentPermissions.findIndex(
+                        (selectedPermission: SelectedPermissionsInterface) => 
+                            selectedPermission.apiResourceId === apiResourceId
+                    );
+
+                    // Add the scope to the existing API resource.
+                    currentPermissions[index].scopes.push({
+                        displayName: permission.display,
+                        id: permission.value,
+                        name: permission.value
+                    });
+                }
+            }
+        });
+        
+        setInitialAPIResourceIds(uniqueAPIResourceIds);
+        setSelectedPermissions(currentPermissions);
+        setInitialSelectedPermissions(currentPermissions);
+    };
+
+    /**
+     * The following function handles the update of the role permissions.
+     */
+    const updateRolePermissions = (): void => {
+        setIsSubmitting(true);
+        const permissionArray: PermissionUpdateInterface[] = [];
+
+        selectedPermissions.map((selectedPermission: SelectedPermissionsInterface) => {
+            selectedPermission.scopes.map((scope: ScopeInterface) => {
+                permissionArray.push({
+                    value: scope.name
+                });
+            });
+        });
```

Comment:
Prefer `forEach` over `map` here, since we are not using the returned array from `map`
```
selectedPermissions.forEach((selectedPermission: SelectedPermissionsInterface) => {
    selectedPermission.scopes.forEach((scope: ScopeInterface) => {
        permissionArray.push({
            value: scope.name
        });
    });
});
```

Reply:
Fixed with ed31181181da4228a981a4e356b5f55ac42ce0db